### PR TITLE
DEVC-1128: Created landing page, profile form and modal for displaying devon instances

### PR DESCRIPTION
**Pull request by suprishi**

### CODE REVIEW

File: main/index.ts [hunk 1]
```diff
@@ -77,6 +92,54 @@ const downloadHandler = (_, item) => {
...
+  new ProfileSetupService()
+    .getProfile()
+    .then((data) => {
+      mainWindow.webContents.send('get:profile', data);
+    })
+    .catch((error) => {
+      mainWindow.webContents.send('get:profile', {});
+    });
+}
+
```
Comment (jambulud) [inline main/index.ts]: Can you extract this to another file/files?

File: renderer/modules/intro/components/LandingPage/IdeDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    display: 'block',
+    padding: '0',
+    fontWeight: 'bold',
+    marginBottom: '32px',
+  },
+  button: {
+    color: '#FFFFFF',
+    borderColor: '#FFFFFF',
+  },
+});
```
Comment (jambulud) [inline renderer/modules/intro/components/LandingPage/IdeDialog.tsx]: Extract this to another file

File: renderer/modules/intro/components/LandingPage/IdeDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        <List>
+          {props.data.map((ide, index) => {
+            return (
+              <ListItem key={index} className={classes.listItem}>
+                <p>Path: {ide.id}</p>
+                <p>Version: {ide.ideConfig.version}</p>
+              </ListItem>
+            );
+          })}
+        </List>
```
Comment (jambulud) [inline renderer/modules/intro/components/LandingPage/IdeDialog.tsx]: Move to a separated component

File: renderer/modules/intro/components/LandingPage/IdeDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      open={props.open}
+    >
+      <DialogTitle id="customized-dialog-title">
+        <Avatar className={classes.avatar}>
+          <PlaylistAddCheckIcon />
+        </Avatar>
+        <p className={classes.titleText}>
+          Devon Dashboard detected already existing devon instances in your
+          system
+        </p>
```
Comment (jambulud) [inline renderer/modules/intro/components/LandingPage/IdeDialog.tsx]: move to another component

File: renderer/modules/intro/components/LandingPage/LandingPage.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    width: '400px',
+  },
+  welcomeText: {
+    width: '400px',
+  },
+  button: {
+    backgroundColor: '#0075B3',
+    color: '#FFFFFF',
+  },
+});
```
Comment (jambulud) [inline renderer/modules/intro/components/LandingPage/LandingPage.tsx]: Move to another file
landing-page.styles.ts

File: renderer/modules/intro/components/ProfileSetup/ProfileSetup.tsx [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+    },
+    '& label.Mui-focused': {
+      padding: '0.2em 0.5em',
+      borderRadius: '4px',
+    },
+    '& .MuiOutlinedInput-root': {
+      backgroundColor: '#FFFFFF',
+    },
+  },
+})(TextField);
```
Comment (jambulud) [inline renderer/modules/intro/components/ProfileSetup/ProfileSetup.tsx]: Move to another file
profile-setup.styles.ts

File: renderer/modules/intro/components/ToolbarContainer/ToolbarContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        backgroundColor: '#0075B3',
+      },
+      logo: {
+        width: 169,
+        height: 32,
+      },
+      pageContent: {
+        minHeight: '100vh',
+      },
+    });
```
Comment (jambulud) [inline renderer/modules/intro/components/ToolbarContainer/ToolbarContainer.tsx]: Move to another file

File: renderer/modules/intro/components/ProfileSetup/ProfileSetup.tsx [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+          variant="contained"
+          color="primary"
+          disabled={checkFormValidity()}
+          onClick={() => createProfile(true)}
+        >
+          Create My Profile
+        </Button>
+        <Button variant="outlined" onClick={() => createProfile(false)}>
+          Will Do It Later
+        </Button>
```
Comment (jambulud) [inline renderer/modules/intro/components/ProfileSetup/ProfileSetup.tsx]: Couldn't each form control be a separated component? 

Each component would receive at least one function (from props) that is executed when the field is resolved?

```tsx
componentForm() {
  const [fields, setFields] = useState({ field1: false, field2: false}) // true if its filled and validated

  const resolveField = (fieldUpdate: { [key: string]: boolean}) => {
    return () =>
      setFields((prevState) => ({
        ...prevState,
        ...fieldUpdate,
      }));
  }

  return (
    <form>
      <FieldComponent1 onResolve={resolveField({ field1: true })}></FieldComponent1>
      <FieldComponent2 onResolve={resolveField({ field2: true })}></FieldComponent2> 
      <Button disabled={fields.values().every((resolved) => resolved)}>Submit form</Button>
    </form>
  )
}

```

**Review by jambulud — CHANGES_REQUESTED** [review]
Rename the files and folders
Instead of CamelCase use snake-case

File: renderer/modules/intro/components/ProfileSetup/ProfileSetup.tsx (hunk 8 above)
Comment (suprishi) [reply renderer/modules/intro/components/ProfileSetup/ProfileSetup.tsx]: Doable. But what does it achieve? Is it for better maintainability or somehow improves performance as well?

Comment (jambulud) [reply renderer/modules/intro/components/ProfileSetup/ProfileSetup.tsx]: Maintanability and readability

When you  have lots of nested tags it is difficult to keep track of what you are reading. Also, isolating smaller components allows you to do modifications in one single formControl (in this case) being sure that you don't break the rest of the form

File: main/index.ts (hunk 1 above)
Comment (suprishi) [reply main/index.ts]: Shall I move them to a new `profile-setup` directory under `main/modules/`

Comment (jambulud) [reply main/index.ts]: Yes, seems to be a good place

**Review by jambulud — APPROVED** [review]

**Pull request merged**